algorithms/kernel/svm/oneapi/svm_train_workset_oneapi.h

```diff
@@ -0,0 +1,215 @@
+/* file: svm_train_workset_oneapi.h */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  SVM workset structure implementation
+//--
+*/
+
+#ifndef __SVM_TRAIN_WORKSET_ONEAPI_H__
+#define __SVM_TRAIN_WORKSET_ONEAPI_H__
+
+#include "service/kernel/service_utils.h"
+#include "algorithms/kernel/svm/oneapi/svm_helper_oneapi.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace svm
+{
+namespace training
+{
+namespace internal
+{
+using namespace daal::services::internal;
+
+template <typename algorithmFPType>
+struct TaskWorkingSet
+{
+    using Helper = utils::internal::HelperSVM<algorithmFPType>;
+
+    TaskWorkingSet(const size_t nVectors) : _nVectors(nVectors) {}
+
+    services::Status init()
+    {
+        services::Status status;
+        auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+        const auto idType = TypeIds::id<algorithmFPType>();
+
+        _sortedFIndices = context.allocate(TypeIds::id<uint32_t>(), _nVectors, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        _indicator = context.allocate(TypeIds::id<uint32_t>(), _nVectors, &status);
+        context.fill(_indicator, 0, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        auto & deviceInfo = context.getInfoDevice();
+
+        const size_t maxWS = deviceInfo.maxWorkGroupSize;
+
+        _nWS       = utils::internal::min(utils::internal::maxpow2(_nVectors), maxWS);
```

Comment:
Shall `_nWS` be always power of two? If yes, I'd recommend to check it for `maxWS` since it's not guaranteed to be power of two. It's most likely be power of two, but I don't very like the fact we depend on hardware configuration here. For example:
```
utils::internal::min(utils::internal::maxpow2(_nVectors), 
                     utils::internal::maxpow2(maxWS));
```

Reply:
Done